src/main/java/org/sagebionetworks/web/client/widget/entity/controller/ProvenanceEditorWidget.java

```diff
@@ -0,0 +1,216 @@
+import org.sagebionetworks.web.client.events.EntityUpdatedHandler;
+import org.sagebionetworks.web.client.widget.entity.browse.EntityFinder;
+
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+public class ProvenanceEditorWidget implements ProvenanceEditorWidgetView.Presenter {
+
+	ProvenanceEditorWidgetView view;
+	SynapseClientAsync synapseClient;
+	SynapseAlert synAlert;
+	PortalGinInjector ginInjector;
+	ProvenanceListWidget usedProvenanceList;
+	ProvenanceListWidget executedProvenanceList;
+	Activity activity;
+	EntityFinder entityFinder;
+	ProvenanceURLDialogWidget urlDialog;
+	EntityUpdatedHandler entityUpdatedHandler;
+
+	
+	@Inject
+	public ProvenanceEditorWidget(ProvenanceEditorWidgetView view,
+			SynapseClientAsync synapseClient, SynapseAlert synAlert,
+			PortalGinInjector ginInjector, EntityFinder entityFinder,
+			ProvenanceURLDialogWidget urlDialog) {
+		this.view = view;
+		this.synapseClient = synapseClient;
+		this.synAlert = synAlert;
+		this.ginInjector = ginInjector;
+		this.entityFinder = entityFinder;
+		this.urlDialog = urlDialog;
+		usedProvenanceList = ginInjector.getProvenanceListWidget();
+		executedProvenanceList = ginInjector.getProvenanceListWidget();
+		usedProvenanceList.setEntityFinder(entityFinder);
+		usedProvenanceList.setURLDialog(urlDialog);
+		executedProvenanceList.setEntityFinder(entityFinder);
+		executedProvenanceList.setURLDialog(urlDialog);
+		view.setSynAlertWidget(synAlert);
+		view.setUsedProvenanceList(usedProvenanceList);
+		view.setExecutedProvenanceList(executedProvenanceList);
+		view.setEntityFinder(entityFinder);
+		view.setURLDialog(urlDialog);
+		view.setPresenter(this);
+	}
+	
+	@Override
+	public void configure(EntityBundle entityBundle, EntityUpdatedHandler entityUpdatedHandler) {
+		this.entityUpdatedHandler = entityUpdatedHandler;
+		clear();
+		Entity entity = entityBundle.getEntity();
+		synapseClient.getOrCreateActivityForEntityVersion(entity.getId(), null, new AsyncCallback<Activity>() {
+			
+			@Override
+			public void onFailure(Throwable caught) {
+				synAlert.handleException(caught);
+			}
+			
+			@Override
+			public void onSuccess(Activity result) {
+				activity = result;
+				view.setName(result.getName());
+				view.setDescription(result.getDescription());
+				Set<Used> allProvenance = result.getUsed();
+				List<ProvenanceEntry> usedEntries = new LinkedList<ProvenanceEntry>();
+				List<ProvenanceEntry> executedEntries = new LinkedList<ProvenanceEntry>();
+				for (Used provEntry: allProvenance) {    
+					ProvenanceEntry toAdd;
+					if (provEntry instanceof UsedEntity) {
+						Reference ref = ((UsedEntity)provEntry).getReference();
+						toAdd = ginInjector.getEntityRefEntry();
+						String entityId = ref.getTargetId();
+						Long version = ref.getTargetVersionNumber();
+						((EntityRefProvEntryView)toAdd).configure(entityId, version != null ? version.toString() : null);
```

Comment:
For readability, please move the ternary operator to a new line, where the result is a descriptive variable name (in this case maybe versionString).  Same with usedURL name on line 99.